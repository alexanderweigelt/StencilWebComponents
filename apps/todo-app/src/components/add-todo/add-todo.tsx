import { FunctionalComponent, h } from '@stencil/core';

interface AddTodoProps {
  handleInputChange: (event: Event) => void;
  addTodo: () => void;
  newTodoText: string;
}

export const AddTodo: FunctionalComponent<AddTodoProps> = ({ handleInputChange, addTodo, newTodoText }) => {
  function handleSubmit(event: Event) {
    event.preventDefault()
    addTodo()
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        id="new-todo"
        placeholder="Add a todo"
        value={newTodoText}
        onInput={handleInputChange}
      />
      <button>Add Todo</button>
    </form>

  );
};



