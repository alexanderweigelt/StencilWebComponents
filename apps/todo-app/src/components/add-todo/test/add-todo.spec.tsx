import { newSpecPage } from '@stencil/core/testing';
import { AddTodo } from '../add-todo';
import { h } from '@stencil/core';

describe('AddToDo Component', () => {
  it('matches the snapshot', async () => {
    const page = await newSpecPage({
      components: [],
      template: () => (
        <AddTodo
          handleInputChange={() => {}}
          addTodo={() => {}}
          newTodoText="Test Todo"
        />
      ),
    });

    expect(page.root).toMatchSnapshot();
  });

  it('calls handleInputChange on input change', async () => {
    const handleInputChangeMock = jest.fn();

    const page = await newSpecPage({
      components: [],
      template: () => (
        <AddTodo
          handleInputChange={handleInputChangeMock}
          addTodo={() => {}}
          newTodoText=""
        />
      ),
    });

    const input = page.root.querySelector('input');
    input.dispatchEvent(new Event('input'));

    expect(handleInputChangeMock).toHaveBeenCalledTimes(1);
  });

  it('calls addTodo on form submit', async () => {
    const addTodoMock = jest.fn();

    const page = await newSpecPage({
      components: [],
      template: () => (
        <AddTodo
          handleInputChange={() => {}}
          addTodo={addTodoMock}
          newTodoText="Test Todo"
        />
      ),
    });

    const form: HTMLFormElement = page.root.querySelector('form');
    form.dispatchEvent(new Event('submit', { cancelable: true }));

    await page.waitForChanges();

    expect(addTodoMock).toHaveBeenCalledTimes(1);
  });
});
