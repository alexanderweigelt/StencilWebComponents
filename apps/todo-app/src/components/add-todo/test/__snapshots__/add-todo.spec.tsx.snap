// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDo Component matches the snapshot 1`] = `
<div class="add-todo">
  <form>
    <input id="new-todo" placeholder="Add a todo" value="Test Todo">
    <button>
      Add Todo
    </button>
  </form>
</div>
`;
